tests/slurm_plugin/test_clustermgtd.py

```diff
@@ -2112,6 +2170,794 @@ def test_update_status(self, compute_fleet_status_manager, put_item_response, ex
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"queue1-st-c5xlarge-1"},
+            None,
+            False,
+        ),
+    ],
+    ids=[
+        "static_self_terminate",
+        "static_timeout",
+        "dynamic_self_terminate",
+        "dynamic_timeout",
+        "dynamic_runinstance",
+        "static_runinstance",
+        "static_joined_cluster",
+        "dynamic_reset_incorrect",
+    ],
+)
+@pytest.mark.usefixtures("initialize_instance_manager_mock", "initialize_compute_fleet_status_manager_mock")
+def test_is_node_bootstrap_failure(
+    node,
+    private_ip_to_instance_map,
+    is_backing_instance_valid,
+    current_node_in_replacement,
+    bootstrap_failure_messages,
+    is_node_bootstrap_failure,
+    mocker,
+    caplog,
+):
+    mocker.patch(
+        "slurm_plugin.clustermgtd.ClusterManager._is_backing_instance_valid", return_value=is_backing_instance_valid
+    )
+    mock_sync_config = SimpleNamespace(node_replacement_timeout=30)
+    caplog.set_level(logging.WARNING)
+    cluster_manager = ClusterManager(mock_sync_config)
+    cluster_manager._current_time = datetime(2020, 1, 2, 0, 0, 0)
+    cluster_manager._static_nodes_in_replacement = current_node_in_replacement
+    # Run tests and assert calls
+    assert_that(cluster_manager._is_node_bootstrap_failure(node, private_ip_to_instance_map)).is_equal_to(
+        is_node_bootstrap_failure
+    )
+    if bootstrap_failure_messages:
+        assert_that(caplog.text).contains(bootstrap_failure_messages)
+
+
+@pytest.mark.parametrize(
+    "nodes, initial_map, expected_map",
+    [
+        (
+            [
+                SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+                SlurmNode("queue2-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue2"),
+            ],
+            {"queue1": 1, "queue2": 1},
+            {"queue1": 2, "queue2": 2},
+        ),
+        (
+            [SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue2")],
+            {"queue1": 1},
+            {"queue1": 1, "queue2": 1},
+        ),
+        (
+            [
+                SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+                SlurmNode("queue1-st-c5large-1", "ip-1", "hostname", "some_state", "queue1"),
+            ],
+            {},
+            {"queue1": 2},
+        ),
+        (
+            [SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1,queue2")],
+            {},
+            {"queue1": 1, "queue2": 1},
+        ),
+    ],
+)
+@pytest.mark.usefixtures("initialize_instance_manager_mock", "initialize_compute_fleet_status_manager_mock")
+def test_increase_partitions_protected_failure_count(nodes, initial_map, expected_map, mocker):
```

Comment:
This function is very straightforward and just increases the count, so we will not get much value by testing it alone. We should test in the caller that partition failure count is as expected when this function is called instead